### ISSUE TITLE
idea: Spellcheck Correction Suggestions and Dictionary Customization

### ISSUE THREAD

**Issue by imtuyethan** [menloresearch/jan#3841]
### Problem Statement
https://discord.com/channels/1107178041848909847/1296752295786184736

This is a feature request to improve the spellcheck functionality in Jan. Users have reported that:
- Spelling errors are underlined, but there are no correction suggestions when right-clicking on the underlined words. The right-click does not trigger any actions.
- There is currently no option to change the dictionary used for spellchecking.

This could help:
- Enhances user experience by providing on-the-fly corrections, improving productivity.
- Allows users to work in their preferred language or with a specialized dictionary for industry-specific terms.

### Feature Idea

1. Spellcheck Suggestions: When a word is underlined due to a spelling error, right-clicking on it should display a list of suggested corrections.
2. Dictionary Customization: Users should be able to change the dictionary used for spellchecks, allowing for multiple languages or custom wordlists.

**Comment by urmauur** [comment]
@imtuyethan Just a heads-up, our current spellcheck is the native one from https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/spellcheck

Additionally, there’s a ticket regarding this issue https://github.com/janhq/jan/issues/1101

Here’s an example of what we need:

<img width="193" alt="Screenshot 2024-10-18 at 22 49 49" src="https://github.com/user-attachments/assets/a3887873-8c98-4f40-9ade-efad4173f100">

**Comment by michaelwschultz** [comment]
I think I have a solution for this but I can't get the app to run locally. Will try again tomorrow and see if I can't get it spun up.

**Comment by louis-menlo** [comment]
Hi @michaelwschultz iIs there any luck with this?

**Comment by michaelwschultz** [comment]
Couldn't get the app working locally to test. I can send you my changes though @louis-menlo. I'll put up a PR and maybe you can test them.

**Comment by michaelwschultz** [comment]
@louis-menlo just posted a PR for ya. https://github.com/menloresearch/jan/pull/4920 Hope this helps and sorry for the delay, been a busy few weeks.

**Comment by louis-menlo** [comment]
This could be addressed when moving to Tauri, then native APIs will work properly. #4844

**Comment by louis-jan** [comment]
Jan now inherits native browser API support, which means all of the menu context items will be available as they should be.
![Image](https://github.com/user-attachments/assets/de4b74c8-a03d-41a6-969f-8e497b25f8f8)
